JExample/trunk/src/main/ch/unibe/jexample/internal/ExampleGraph.java

```diff
@@ -102,11 +102,22 @@ public Example findExample(MethodReference ref) {
     public void run(ExampleClass testClass, RunNotifier notifier) {
         if (!anyHasBeenRun) onStartRunning();
         anyHasBeenRun = true;
+                
+        for(Method m : testClass.getImplementingClass().getMethods()){

```

Comment:
As far as I remember `@BeforeClass` functions are run lazily in `Example#bareInvoke` when the first example function of an example class is run. Did you try that?